### ISSUE TITLE
Not able to resolve a simple DID after creating it

### ISSUE THREAD

**Issue by Muhammad-Altabba** [transmute-industries/sidetree.js#234]
Hello,
I have the following code that creates a DID and resolves it after. The creation seems to be ok. But the resolution fails. Here is the code and its output:

```javascript
// Create a long-form DID string.
const createOperationData = await OperationGenerator.generateCreateOperation();
const didMethodName = 'sidetree';
const didUniqueSuffix =
  createOperationData.createOperation.didUniqueSuffix;
const shortFormDid = `did:${didMethodName}:${didUniqueSuffix}`;
const encodedSuffixData =
  createOperationData.createOperation.encodedSuffixData;
const encodedDelta = createOperationData.createOperation.encodedDelta;
const longFormDid = `${shortFormDid}?-sidetree-initial-state=${encodedSuffixData}.${encodedDelta}`;

console.log(longFormDid)

// Resove the created DID
const resolver = new Resolver(element.versionManager, element.operationStore);
const didState = (await resolver.resolve(didUniqueSuffix)) as DidState;

console.log('didState', didState)
const didDoc = didState.document;
console.log('didDoc',didDoc)
```

Here is the output:

> Long Form Did : 
>  did:sidetree:EiDZcJL-9ykz4p4u6NnU8iNvsL0u2DPX8Vtp1cgEa3iILw?-sidetree-initial-state=eyJkZWx0YV9oYXNoIjoiRWlBQzgxeVJHU0xib001N29faWU4WENQVTJyVVdyWTVLWDBzQjNzRnI1eTlCdyIsInJlY292ZXJ5X2NvbW1pdG1lbnQiOiJFaURWMEJ0V0tXV2dJNTlDUWFWMVpYb0NvQjU4UzQ1VV9NNHFjTnVqQk9wa1l3In0.eyJ1cGRhdGVfY29tbWl0bWVudCI6IkVpQnRfS3FOZ1VRa1l6U0JhRkxfcjlBc3NOZENSZ2p5aXdfZGdNckc4bDZheFEiLCJwYXRjaGVzIjpbeyJhY3Rpb24iOiJyZXBsYWNlIiwiZG9jdW1lbnQiOnsicHVibGljX2tleXMiOlt7ImlkIjoic2lnbmluZ0tleSIsInR5cGUiOiJFZDI1NTE5VmVyaWZpY2F0aW9uS2V5MjAxOCIsImp3ayI6eyJjcnYiOiJFZDI1NTE5IiwieCI6InJLR01ZQ3NqTmUwVmpDMWlZdUhKSExTMmJlYnZ0N2F4OXpIMkVjVTJCUVkiLCJrdHkiOiJPS1AiLCJraWQiOiJPVFB3LWMxemNsNUFiTVlROFdrbi0wMTNrSGV6RWNDNDVaaTJVTFd4VDFzIn0sInB1cnBvc2UiOlsiYXV0aCIsImdlbmVyYWwiLCJhZ3JlZW1lbnQiLCJhc3NlcnRpb24iLCJkZWxlZ2F0aW9uIiwiaW52b2NhdGlvbiJdfV0sInNlcnZpY2VfZW5kcG9pbnRzIjpbeyJpZCI6InNlcnZpY2VFbmRwb2ludElkMTIzIiwidHlwZSI6InNvbWVUeXBlIiwiZW5kcG9pbnQiOiJodHRwczovL3d3dy51cmwuY29tIn1dfX1dfQ
> Resolving DID unique suffix 'EiDZcJL-9ykz4p4u6NnU8iNvsL0u2DPX8Vtp1cgEa3iILw'...
> didState =  undefined
> (node:14910) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'document' of undefined

Any ideas and suggestions are welcomed.

Thanks in advance,

**Comment by Muhammad-Altabba** [comment]
I found it. The `generateCreateOperation` will only create an in-memory JSON object. 
So, before trying to resolve the new DID, the `handleOperationRequest` and `triggerBatchAndObserve` has to be called in order to actually create the DID and push it to the sidetree... 



```
  const responseOfOperation = await element.handleOperationRequest(
    Buffer.from(JSON.stringify(createOperationData.operationRequest))
  );
  await element.triggerBatchAndObserve();

```

**Comment by Muhammad-Altabba** [comment]
I found it. The `generateCreateOperation` will only create an in-memory JSON object. 
So, before trying to resolve the new DID, the `handleOperationRequest` and `triggerBatchAndObserve` have to be called in order to actually create the DID and push it to the sidetree... 



```
  const responseOfOperation = await element.handleOperationRequest(
    Buffer.from(JSON.stringify(createOperationData.operationRequest))
  );
  await element.triggerBatchAndObserve();

```

However, I tried the same code many times and I put it in a `for` loop. It succeed with about only 15 times out of 100 tires. 

Any ideas or suggesions?

Thanks in advance,
